src/java/org/apache/cassandra/service/consensus/migration/ConsensusMigrationRepairResult.java

```diff
@@ -33,18 +34,30 @@ private ConsensusMigrationRepairResult(ConsensusMigrationRepairType type, Epoch
         this.minEpoch = minEpoch;
     }
 
-    public static ConsensusMigrationRepairResult fromCassandraRepair(Epoch minEpoch, boolean migrationEligibleRepair)
+    public static ConsensusMigrationRepairResult fromRepair(Epoch minEpoch, boolean paxosRepaired, boolean accordRepaired, boolean deadNodesExcluded)
     {
-        checkArgument(!migrationEligibleRepair || minEpoch.isAfter(Epoch.EMPTY), "Epoch should not be empty if Paxos and regular repairs were performed");
-        if (migrationEligibleRepair)
-            return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.paxos, minEpoch);
-        else
-            return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.ineligible, Epoch.EMPTY);
+        checkArgument((!paxosRepaired && !accordRepaired) || minEpoch.isAfter(Epoch.EMPTY), "Epoch should not be empty if Paxos and regular repairs were performed");
+
+        if (deadNodesExcluded) return INELIGIBLE;
+        if (paxosRepaired && accordRepaired) return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.either, minEpoch);
+        if (paxosRepaired) return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.paxos, minEpoch);
+        if (accordRepaired) return new ConsensusMigrationRepairResult(ConsensusMigrationRepairType.accord, minEpoch);
+        return INELIGIBLE;
+    }
+
+    public static ConsensusMigrationRepairResult fromPaxosOnlyRepair(Epoch minEpoch, boolean deadNodesExcluded)
+    {
+        return fromRepair(minEpoch, true, false, deadNodesExcluded);
```

Comment:
I think this is mixing up how Paxos repair needs to work unless I am wrong. Paxos repair doesn't finish at `ALL` and the requirement for Accord for migration is that it be done at `ALL` so that eventually Accord can safely switch to single replica reads.

That is why previously the Paxos only repair was not marked as migration eligible.

If Paxos repair actually applies everything at all then this would work, but I recall that it doesn't because that wouldn't be highly available.

Reply:
good catch, thanks. Fixed